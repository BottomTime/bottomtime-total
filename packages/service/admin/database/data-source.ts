import path from 'path';
import { DataSource } from 'typeorm';

export async function getDataSource(
  url: string,
  requireSsl: boolean,
): Promise<DataSource> {
  const client = new DataSource({
    type: 'postgres',
    url,
    entities: [path.resolve(__dirname, '../../src/data/**/*.entity.ts')],
    ssl: requireSsl,
  });
  return await client.initialize();
}
