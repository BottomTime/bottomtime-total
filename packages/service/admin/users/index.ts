import { CommandModule } from 'yargs';

import { createAdmin } from './create-admin';
import { getUserToken } from './get-user-token';

export const userModule: CommandModule<{
  'postgres-uri': string;
  'require-ssl': boolean;
}> = {
  command: 'user',

  describe: 'Commands for managing a user account',

  builder(yargs) {
    return yargs
      .command(
        'new-admin',
        'Creates a new admin user.',
        (yargs) => {
          return yargs
            .positional('username', {
              alias: 'u',
              description: "The user's username.",
              type: 'string',
            })
            .positional('password', {
              alias: 'p',
              description: 'The password to set on the new admin account',
              type: 'string',
            })
            .help();
        },
        async (yargs) => {
          await createAdmin({
            postgresRequireSsl: yargs.requireSsl,
            potgresUri: yargs.postgresUri,
            username: yargs.username,
            password: yargs.password,
          });
        },
      )
      .command(
        'token <user>',
        'Gets a token for a user',
        (yargs) => {
          return yargs
            .positional('user', {
              demandOption: true,
              description: "The user's username or email address.",
              type: 'string',
            })
            .help();
        },
        async (yargs) => {
          await getUserToken(yargs.postgresUri, yargs.requireSsl, yargs.user);
        },
      )
      .help();
  },

  async handler() {
    /* TODO */
  },
};
