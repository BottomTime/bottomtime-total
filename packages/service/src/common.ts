import { z } from 'zod';
import { DepthUnit } from './constants';

export const BooleanString = z
  .enum(['true', 'false'])
  .transform((value) => value === 'true');

export const DepthSchema = z.object({
  depth: z.number().min(0),
  unit: z.nativeEnum(DepthUnit),
});
export type Depth = z.infer<typeof DepthSchema>;

export const GpsCoordinatesSchema = z.object({
  lat: z.number().gte(-90).lte(90),
  lon: z.number().gte(-180).lte(180),
});
export type GpsCoordinates = z.infer<typeof GpsCoordinatesSchema>;

export interface Range {
  min: number;
  max: number;
}
