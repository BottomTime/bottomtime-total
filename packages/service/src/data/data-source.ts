import path from 'path';
import { DataSource, DataSourceOptions } from 'typeorm';

import { Config } from '../config';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: [path.join(__dirname, './**/*.entity.ts')],
  synchronize: false,
};

export async function initDataSource(): Promise<DataSource> {
  const AppDataSource = new DataSource(PostgresDataSourceOptions);
  return await AppDataSource.initialize();
}
