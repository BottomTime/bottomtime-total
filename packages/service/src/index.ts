import { Config } from './config';
import { createApp } from './create-app';
import { createLogger } from './logger';

const log = createLogger(Config.logLevel);

createApp(log)
  .then(async (app) => {
    await app.listen(Config.port);

    log.info(
      `ðŸŽ‰ Service has successfully started and is listening on port ${Config.port}. ðŸŽ‰`,
    );
  })
  .catch((error) => {
    log.fatal('[SERVICE] Failed to start service... Shutting down...', error);
    process.exit(1);
  });
