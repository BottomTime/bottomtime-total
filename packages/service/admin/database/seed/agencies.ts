import { AgencyEntity } from '../../../src/data';

export const SeedAgencies: AgencyEntity[] = [
  {
    id: '5ab222fa-3a71-4dbb-908a-92f281c16705',
    name: 'PADI',
    longName: 'Professional Association of Diving Instructors',
    logo: '/img/agencies/padi.png',
    website: 'https://www.padi.com',
    ordinal: 1,
  },
  {
    id: '7db437c8-4336-4a7b-b5b8-778f3c69905d',
    name: 'SSI',
    longName: 'Scuba Schools International',
    logo: '/img/agencies/ssi.png',
    website: 'https://www.divessi.com',
    ordinal: 2,
  },
  {
    id: '384d9fcb-6050-4760-a441-abc3e22d7e6e',
    name: 'CMAS',
    longName: 'Confédération Mondiale des Activités Subaquatiques',
    logo: '/img/agencies/cmas.png',
    website: 'https://www.cmas.org',
    ordinal: 3,
  },
  {
    id: '4967e7e8-dae1-43f8-88d5-f26aa273aba2',
    name: 'NAUI',
    longName: 'National Association of Underwater Instructors',
    logo: '/img/agencies/naui.png',
    website: 'https://www.naui.org',
    ordinal: 4,
  },
  {
    id: '04b4cd16-fceb-4692-8acf-cf2d15fac0b5',
    name: 'BSAC',
    longName: 'British Sub-Aqua Club',
    logo: '/img/agencies/bsac.png',
    website: 'https://www.bsac.com',
    ordinal: 5,
  },
  {
    id: '0f409465-5d1e-46c6-afcb-ab68843b7ec0',
    name: 'SDI',
    longName: 'Scuba Diving International',
    logo: '/img/agencies/sdi.png',
    website: 'https://www.tdisdi.com',
    ordinal: 6,
  },
  {
    id: '8acbfb2d-9546-4c2a-82b5-31b675fdcef8',
    name: 'RAID',
    longName: 'Rebreather Association of International Divers',
    logo: '/img/agencies/raid.png',
    website: 'https://www.diveraid.com',
    ordinal: 7,
  },
  {
    id: 'fc1087d3-b58e-4664-9f1c-899a93439432',
    name: 'TDI',
    longName: 'Technical Diving International',
    logo: '/img/agencies/tdi.png',
    website: 'https://www.tdisdi.com',
    ordinal: 8,
  },
];
