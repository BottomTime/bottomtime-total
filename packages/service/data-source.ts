import path from 'path';
import { DataSource } from 'typeorm';

import { PostgresDataSourceOptions } from './src/data/data-source';

export default new DataSource({
  ...PostgresDataSourceOptions,
  migrations: [path.join(__dirname, './migrations/*.ts')],
});
