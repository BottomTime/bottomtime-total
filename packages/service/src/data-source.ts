import 'reflect-metadata';
import { DataSourceOptions } from 'typeorm';

import { Config } from './config';
import * as Entities from './data';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: Object.values(Entities),
  synchronize: false,
};
