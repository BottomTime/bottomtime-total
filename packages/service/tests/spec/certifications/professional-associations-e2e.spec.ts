import {
  CreateOrUpdateProfessionalAssociationParamsDTO,
  UserRole,
} from '@bottomtime/api';

import { HttpServer, INestApplication } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';

import request from 'supertest';
import { Repository } from 'typeorm';

import {
  AgenciesService,
  AgencyFactory,
  ProfessionalAssociationsService,
} from '../../../src/certifications';
import { ProfessionalAssociationsController } from '../../../src/certifications/professional-associations.controller';
import {
  AgencyEntity,
  UserEntity,
  UserProfessionalAssociationsEntity,
} from '../../../src/data';
import { UsersModule } from '../../../src/users';
import { dataSource } from '../../data-source';
import { TestAgencies } from '../../fixtures/agencies';
import { createAuthHeader, createTestApp, createTestUser } from '../../utils';

const Username = 'joe.regular';

function getUrl(associationId?: string, username?: string): string {
  const url = `/api/users/${username ?? Username}/professionalAssociations`;
  return associationId ? `${url}/${associationId}` : url;
}

const TestData: UserProfessionalAssociationsEntity[] = [
  {
    id: '7660e6de-d97e-4819-a15e-c7751f8690dc',
    identificationNumber: 'CK-13333',
    startDate: '2021-01-01',
    title: 'Instructor',
    agency: TestAgencies[0],
  },
  {
    id: 'b77ad1ca-0cd9-448b-9be5-b829d9024aa4',
    identificationNumber: 'abd2324',
    startDate: '2022-01-01',
    title: 'Dive Master',
    agency: TestAgencies[1],
  },
  {
    id: '70be1a4b-15cb-4ede-b91f-5c3c02280bf4',
    identificationNumber: '123456789',
    startDate: '2023-01-01',
    title: 'Instructor',
    agency: TestAgencies[2],
  },
];

describe('Professional associations E2E tests', () => {
  let Users: Repository<UserEntity>;
  let Agencies: Repository<AgencyEntity>;
  let Associations: Repository<UserProfessionalAssociationsEntity>;

  let app: INestApplication;
  let server: HttpServer;

  let userData: UserEntity;
  let otherUserData: UserEntity;
  let adminUserData: UserEntity;

  let authHeader: [string, string];
  let otherAuthHeader: [string, string];
  let adminAuthHeader: [string, string];

  let testData: UserProfessionalAssociationsEntity[];

  beforeAll(async () => {
    Users = dataSource.getRepository(UserEntity);
    Agencies = dataSource.getRepository(AgencyEntity);
    Associations = dataSource.getRepository(UserProfessionalAssociationsEntity);

    app = await createTestApp({
      imports: [
        TypeOrmModule.forFeature([
          AgencyEntity,
          UserProfessionalAssociationsEntity,
        ]),
        UsersModule,
      ],
      providers: [
        AgenciesService,
        AgencyFactory,
        ProfessionalAssociationsService,
      ],
      controllers: [ProfessionalAssociationsController],
    });
    await app.init();
    server = app.getHttpServer();

    userData = createTestUser({ username: Username });
    otherUserData = createTestUser();
    adminUserData = createTestUser({ role: UserRole.Admin });

    [authHeader, otherAuthHeader, adminAuthHeader] = await Promise.all([
      createAuthHeader(userData.id),
      createAuthHeader(otherUserData.id),
      createAuthHeader(adminUserData.id),
    ]);
    testData = [
      {
        ...TestData[0],
        user: userData,
      },
      {
        ...TestData[1],
        user: userData,
      },
      {
        ...TestData[2],
        user: otherUserData,
      },
    ];
  });

  beforeEach(async () => {
    await Users.save([userData, otherUserData, adminUserData]);
    await Agencies.save(TestAgencies);
  });

  afterAll(async () => {
    await app.close();
  });

  describe('when listing associations', () => {
    beforeEach(async () => {
      await Associations.save(testData);
    });

    it('will list professional associations', async () => {
      const { body } = await request(server)
        .get(getUrl())
        .set(...authHeader)
        .expect(200);

      expect(body).toMatchSnapshot();
    });

    it('will return a 404 response if the user does not exist', async () => {
      await request(server)
        .get(getUrl(undefined, 'not.a.user'))
        .set(...adminAuthHeader)
        .expect(404);
    });
  });

  describe('when retrieving a single association', () => {
    beforeEach(async () => {
      await Associations.save(testData);
    });

    it('will return the indicated association', async () => {
      const { body } = await request(server)
        .get(getUrl(testData[0].id))
        .set(...authHeader)
        .expect(200);

      expect(body).toMatchSnapshot();
    });

    it('will return a 404 response if the association does not exist', async () => {
      await request(server)
        .get(getUrl('8032ea62-06e2-4340-9c91-a3c39c1de234'))
        .expect(404);
    });

    it('will return a 404 response if the user does not exist', async () => {
      await request(server)
        .get(getUrl(testData[0].id, 'not.a.user'))
        .expect(404);
    });

    it('will return a 404 response if the indicated association does not belong to the user', async () => {
      await request(server)
        .get(getUrl(testData[2].id))
        .set(...authHeader)
        .expect(404);
    });
  });

  describe('when creating a new association', () => {
    const options: CreateOrUpdateProfessionalAssociationParamsDTO = {
      agency: TestAgencies[1].id,
      identificationNumber: 'MY-248992',
      startDate: '2022-08-18',
      title: 'Instructor',
    };

    it('will create a new association', async () => {
      const { body } = await request(server)
        .post(getUrl())
        .set(...authHeader)
        .send(options)
        .expect(201);

      expect(body).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });

      const saved = await Associations.findOneOrFail({
        where: { id: body.id },
        relations: ['agency'],
      });
      expect(saved).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
          ordinal: TestAgencies[1].ordinal,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });
    });

    it('will allow an admin to create a new association', async () => {
      const { body } = await request(server)
        .post(getUrl())
        .set(...adminAuthHeader)
        .send(options)
        .expect(201);

      expect(body).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });

      const saved = await Associations.findOneOrFail({
        where: { id: body.id },
        relations: ['agency'],
      });
      expect(saved).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
          ordinal: TestAgencies[1].ordinal,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });
    });

    it('will return a 400 response if the request body is invalid', async () => {
      await request(server)
        .post(getUrl())
        .set(...authHeader)
        .send({
          identificationNumber: 77,
          startDate: 'yesterday',
          title: true,
          agency: TestAgencies[0].id,
        })
        .expect(400);
    });

    it('will return a 400 resonse if the request body is missing', async () => {
      await request(server)
        .post(getUrl())
        .set(...authHeader)
        .expect(400);
    });

    it('will return a 400 response if the indicated agency does not exist', async () => {
      await request(server)
        .post(getUrl())
        .set(...authHeader)
        .send({
          ...options,
          agency: '71721872-8b56-465b-a1b5-2656c822ad15',
        })
        .expect(400);
    });

    it('will return a 401 response if the request is unauthenticated', async () => {
      await request(server).post(getUrl()).send(options).expect(401);
    });

    it('will return a 403 response if the user is not authorized to create the association', async () => {
      await request(server)
        .post(getUrl())
        .set(...otherAuthHeader)
        .send(options)
        .expect(403);
    });

    it('will return a 404 response if the user does not exist', async () => {
      await request(server)
        .post(getUrl(undefined, 'not.a.user'))
        .set(...adminAuthHeader)
        .send(options)
        .expect(404);
    });
  });

  describe('when deleting an association', () => {
    beforeEach(async () => {
      await Associations.save(testData[0]);
    });

    it('will delete the indicated association', async () => {
      await request(server)
        .delete(getUrl(testData[0].id))
        .set(...authHeader)
        .expect(204);

      await expect(
        Associations.findOneBy({ id: testData[0].id }),
      ).resolves.toBeNull();
    });

    it('will allow an admin to delete an association', async () => {
      await request(server)
        .delete(getUrl(testData[0].id))
        .set(...adminAuthHeader)
        .expect(204);

      await expect(
        Associations.findOneBy({ id: testData[0].id }),
      ).resolves.toBeNull();
    });

    it('will return a 401 response if the request is unauthenticated', async () => {
      await request(server).delete(getUrl(testData[0].id)).expect(401);
    });

    it('will return a 403 response if the user is not authorized to delete the association', async () => {
      await request(server)
        .delete(getUrl(testData[0].id))
        .set(...otherAuthHeader)
        .expect(403);
    });

    it('will return a 404 response if the association does not exist', async () => {
      await request(server)
        .delete(getUrl('4a2d1774-d77a-4830-977f-af941b162e65'))
        .set(...authHeader)
        .expect(404);
    });

    it('will return a 404 response if the user does not exist', async () => {
      await request(server)
        .delete(getUrl(testData[0].id, 'not.a.user'))
        .set(...adminAuthHeader)
        .expect(404);
    });

    it('will return a 404 response if the association belongs to another user', async () => {
      await request(server)
        .delete(getUrl(testData[2].id))
        .set(...adminAuthHeader)
        .expect(404);
    });
  });

  describe('when updating an existing professional association', () => {
    const options: CreateOrUpdateProfessionalAssociationParamsDTO = {
      agency: TestAgencies[1].id,
      identificationNumber: 'MY-248992',
      startDate: '2022-08-18',
      title: 'Instructor',
    };

    beforeEach(async () => {
      await Associations.save(testData);
    });

    it('will update an existing association', async () => {
      const { body } = await request(server)
        .put(getUrl(testData[0].id))
        .set(...authHeader)
        .send(options)
        .expect(200);

      expect(body).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });

      const saved = await Associations.findOneOrFail({
        where: { id: body.id },
        relations: ['agency'],
      });
      expect(saved).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
          ordinal: TestAgencies[1].ordinal,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });
    });

    it('will allow an admin to update an association', async () => {
      const { body } = await request(server)
        .put(getUrl(testData[0].id))
        .set(...adminAuthHeader)
        .send(options)
        .expect(200);

      expect(body).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });

      const saved = await Associations.findOneOrFail({
        where: { id: body.id },
        relations: ['agency'],
      });
      expect(saved).toEqual({
        agency: {
          id: TestAgencies[1].id,
          logo: TestAgencies[1].logo,
          longName: TestAgencies[1].longName,
          name: TestAgencies[1].name,
          website: TestAgencies[1].website,
          ordinal: TestAgencies[1].ordinal,
        },
        id: body.id,
        identificationNumber: options.identificationNumber,
        startDate: options.startDate,
        title: options.title,
      });
    });

    it('will return a 400 response if the request body is missing', async () => {
      await request(server)
        .put(getUrl(testData[0].id))
        .set(...authHeader)
        .expect(400);
    });

    it('will return a 400 response if the request body is invalid', async () => {
      await request(server)
        .put(getUrl(testData[0].id))
        .set(...authHeader)
        .send({
          identificationNumber: 77,
          startDate: 'yesterday',
          title: true,
          agency: TestAgencies[0].id,
        })
        .expect(400);
    });

    it('will return a 400 response if the agency does not exist', async () => {
      await request(server)
        .put(getUrl(testData[0].id))
        .set(...authHeader)
        .send({
          ...options,
          agency: '71721872-8b56-465b-a1b5-2656c822ad15',
        })
        .expect(400);
    });

    it('will return a 401 response if the request is unauthenticated', async () => {
      await request(server)
        .put(getUrl(testData[0].id))
        .send(options)
        .expect(401);
    });

    it('will return a 403 response if the user is not authorized to update the association', async () => {
      await request(server)
        .put(getUrl(testData[0].id))
        .set(...otherAuthHeader)
        .send(options)
        .expect(403);
    });

    it('will return a 404 response if the association does not exist', async () => {
      await request(server)
        .put(getUrl('0295324b-5b52-4602-9604-33a45dfb4455'))
        .set(...authHeader)
        .send(options)
        .expect(404);
    });

    it('will return a 404 response if the user does not exist', async () => {
      await request(server)
        .put(getUrl(testData[0].id, 'not.a.user'))
        .set(...adminAuthHeader)
        .send(options)
        .expect(404);
    });

    it('will return a 404 response if the association belongs to another user', async () => {
      await request(server)
        .put(getUrl(testData[2].id))
        .set(...adminAuthHeader)
        .send(options)
        .expect(404);
    });
  });
});
