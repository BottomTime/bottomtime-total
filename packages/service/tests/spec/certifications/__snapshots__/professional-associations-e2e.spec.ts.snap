// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Professional associations E2E tests when listing associations will list professional associations 1`] = `
{
  "data": [
    {
      "agency": {
        "id": "aa170980-9a68-4168-9859-1f71e0d023df",
        "logo": "https://www.padi.com",
        "longName": "Professional Association of Diving Instructors",
        "name": "PADI",
        "website": "https://www.padi.com",
      },
      "id": "7660e6de-d97e-4819-a15e-c7751f8690dc",
      "identificationNumber": "CK-13333",
      "startDate": "2021-01-01",
      "title": "Instructor",
    },
    {
      "agency": {
        "id": "d85b955b-04f3-4803-ba74-c31c27bc26de",
        "logo": "https://www.naui.org",
        "longName": "National Association of Underwater Instructors",
        "name": "NAUI",
        "website": "https://www.naui.org",
      },
      "id": "b77ad1ca-0cd9-448b-9be5-b829d9024aa4",
      "identificationNumber": "abd2324",
      "startDate": "2022-01-01",
      "title": "Dive Master",
    },
  ],
  "totalCount": 2,
}
`;

exports[`Professional associations E2E tests when retrieving a single association will return the indicated association 1`] = `
{
  "agency": {
    "id": "aa170980-9a68-4168-9859-1f71e0d023df",
    "logo": "https://www.padi.com",
    "longName": "Professional Association of Diving Instructors",
    "name": "PADI",
    "website": "https://www.padi.com",
  },
  "id": "7660e6de-d97e-4819-a15e-c7751f8690dc",
  "identificationNumber": "CK-13333",
  "startDate": "2021-01-01",
  "title": "Instructor",
}
`;
